### PR TITLE
Feature/integrate new models

### PR DESCRIPTION
**Pull request by krusche**
### Checklist
- [x] I run `yarn run webpack:build:main`: the project builds without errors.
- [x] I run `yarn lint`: there project builds without code style warnings.
- [x] I removed unnecessary whitespace changes in all related files.
- [x] I updated the documentation and models.
- [x] I tested the changes and all related features on the test server https://artemistest.ase.in.tum.de.
- [x] ~I added (end-to-end) test cases for the new functionality.~

### Motivation and Context
We need database changes for new features: notifications, question and answers and lecture support.
In addition, I refactored questions into quiz-questions and fixed some bugs, e.g. that SA questions send the solution as part of the http response to the client before the quiz ends and that the practice mode includes the solution as part of the http response before the quiz ends 

### Description
- [x] Added the entities to artemis.jh for discussion
- [x] Created the Java domain classes (Server)
- [x] Created the liquibase changelog
- [x] Created the TypeScript domain classes (Client)
 
### Steps for Testing
1. Start the server
2. Start the client
3. Check that there are no error messages

### CODE REVIEW

**Comment by maximilianmeier** [comment]
do we need to integrate `lastNotificationRead` from the `User` class to the `artemis.jh` file?

File: artemis.jh [hunk 1]
```diff
@@ -335,12 +338,98 @@ enum TutorParticipationStatus {
...
 }
 
+entity StudentQuestion {
+    questionText String,               // VarChar(1000)
+    creationDate ZonedDateTime,
+    visibleForStudents Boolean
+}
+
+entity QuestionAnswer {
+    answerText String,                 // LONGTEXT --> lob
```
Comment (jpbernius) [inline artemis.jh]: The `LONGTEXT` is not reflected in the migration!

File: artemis.jh [hunk 2]
```diff
@@ -335,12 +338,98 @@ enum TutorParticipationStatus {
     COMPLETED
 }
 
+entity StudentQuestion {
+    questionText String,               // VarChar(1000)
```
Comment (jpbernius) [inline artemis.jh]: The `VarChar(1000)` is not reflected in the migration!

File: artemis.jh [hunk 3]
```diff
@@ -335,12 +338,98 @@ enum TutorParticipationStatus {
...
+}
+
+enum Language {
+    ENGLISH,
+    GERMAN
+}
+
+entity Notification {
+    title String,
+    text String,                       // LONGTEXT --> lob
```
Comment (jpbernius) [inline artemis.jh]: The `LONGTEXT` is not reflected in the migration!

File: artemis.jh [hunk 4]
```diff
@@ -335,12 +338,98 @@ enum TutorParticipationStatus {
...
+}
+
+enum SystemNotificationType {
+    WARNING,
+    INFO
+}
+
+entity Lecture {
+    title String,
+    description String,               // LONGTEXT --> lob
```
Comment (jpbernius) [inline artemis.jh]: The `LONGTEXT` is not reflected in the migration!

**Pull request merged**